scripts/temporal_memory_performance_benchmark.py

```diff
@@ -20,175 +20,380 @@
-            permanenceDec=0.05,
-            activationThreshold=15,
-            globalDecay=0, burnIn=1,
-            checkSynapseConsistency=False,
-            pamLength=1)
-
-    tp10x2 = TP10X2(numberOfCols=2048,
-                    cellsPerColumn=32,
-                    initialPerm=0.5,
-                    connectedPerm=0.8,
-                    minThreshold=10,
-                    newSynapseCount=12,
-                    permanenceInc=0.1,
-                    permanenceDec=0.05,
-                    activationThreshold=15,
-                    globalDecay=0, burnIn=1,
-                    checkSynapseConsistency=False,
-                    pamLength=1)
-
-    def tmComputeFn(pattern, instance):
-      instance.compute(pattern, True)
-
-    def tpComputeFn(pattern, instance):
-      array = cls._patternToNumpyArray(pattern)
-      instance.compute(array, enableLearn=True, computeInfOutput=True)
-
-    return (
-        ("TM (py)", tmPy, tmComputeFn),
-        ("TM (C++)", tmCPP, tmComputeFn),
-        ("TP", tp, tpComputeFn),
-        ("TP10X2", tp10x2, tpComputeFn),
-    )
-
-
-  @classmethod
-  def runAll(cls):
-    impls = cls.setUp()
-    sequence = cls._generateSequence()
-    times = cls._feedAll(impls, sequence)
-    return times
-
-
-  @staticmethod
-  def _generateSequence():
-    scalarEncoder = RandomDistributedScalarEncoder(0.88)
-    sequence = []
-    with open (_INPUT_FILE_PATH) as fin:
-      reader = csv.reader(fin)
-      reader.next()
-      reader.next()
-      reader.next()
-      for _ in xrange(NUM_PATTERNS):
-        record = reader.next()
-        value = float(record[1])
-        encodedValue = scalarEncoder.encode(value)
-        activeBits = set(encodedValue.nonzero()[0])
-        sequence.append(activeBits)
-    return sequence
-
-
-  @classmethod
-  def _feedAll(cls, impls, sequence):
-    repeatedSequence = sequence
-
-    times = collections.defaultdict(float)
-
-    for patNum, pattern in enumerate(repeatedSequence):
-      for name, impl, fn in impls:
-        times[name] += cls._feedOne(pattern, impl, fn)
-      cls._printProgressBar(patNum, len(repeatedSequence), 50)
-
-    return times
-
-
-  @staticmethod
-  def _feedOne(pattern, instance, computeFn):
-    start = time.clock()
-
-    if pattern == None:
-      instance.reset()
-    else:
-      computeFn(pattern, instance)
-
-    elapsed = time.clock() - start
-
-    return elapsed
-
-
-  @staticmethod
-  def _patternToNumpyArray(pattern):
-    array = numpy.zeros(2048, dtype='int32')
-    array[list(pattern)] = 1
-
-    return array
-
-
-  @staticmethod
-  def _printProgressBar(completed, total, nDots):
-    def numberOfDots(n):
-      return (n * nDots) // total
-    completedDots = numberOfDots(completed)
-    if completedDots != numberOfDots(completed - 1):
-      print "\r|" + ("." * completedDots) + (" " * (nDots - completedDots)) + "|",
-      sys.stdout.flush()
 
+  def addContestant(self, constructor, paramsFn, computeFn, name):
```

Comment:
Then the "contestant" would only be usable once.

Also, different tests use different params.

I'm trying to think of ways this could be rearranged, but every option I can think of would require a paramsFn to be used.